Search filters on mobile

leoshatrushin opened the pull request:
Fix for feature request #7650 by adding a funnel icon to the search bar which opens up a modal were the filters can be set

Note: I don't know how to test the search functionality, so I can't ensure I haven't messed it up

Possible points of discussion:
- The specific funnel icon used
- Whether the funnel icon should disappear for large screen sizes

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205826751955553) by [Unito](https://www.unito.io)

oetherington commented on packages/lesswrong/components/search/SearchFilters.tsx:
```diff
@@ -0,0 +1,200 @@
+import { Link } from '../../lib/reactRouterWrapper';
+import Select from '@material-ui/core/Select';
+import {
+    ElasticSorting,
```
```suggestion
  ElasticSorting,
```

Mis-matched indentation here

oetherington reviewed (changes requested):
Thanks - looks really good! I've left a couple of small suggestion comments on the PR.

WRT the icon, normally the EA Forum uses Heroicons and LessWrong uses MUI icons. It looks like the custom SVG you added is actually the filter icon from Heroicons, so we can just
```
import FunnelIcon from "@heroicons/react/24/outline/FunnelIcon";
```
instead of adding the custom SVG. The MUI filter icon looks pretty bad in my opinion, so I'd probably just stick with heroicons for both sites unless LW object.

FYI @agnesstenlund and @b0b3rt 

<img width="651" alt="Screenshot 2023-10-30 at 17 06 08" src="https://github.com/ForumMagnum/ForumMagnum/assets/5075628/553c7b93-daa4-45c7-9653-8d229e2c8660">
<img width="651" alt="Screenshot 2023-10-30 at 17 06 10" src="https://github.com/ForumMagnum/ForumMagnum/assets/5075628/e78200f2-3518-4afa-8106-a8f318672603">
<img width="645" alt="Screenshot 2023-10-30 at 17 05 56" src="https://github.com/ForumMagnum/ForumMagnum/assets/5075628/288bb6f7-0f7b-4238-86f4-378a5c21808e">
<img width="643" alt="Screenshot 2023-10-30 at 17 05 58" src="https://github.com/ForumMagnum/ForumMagnum/assets/5075628/b4f1f45d-51c4-48df-8b44-5a899adfafff">
<img width="327" alt="Screenshot 2023-10-30 at 17 06 23" src="https://github.com/ForumMagnum/ForumMagnum/assets/5075628/6a82c1fe-3b62-460c-860a-f7e22da17a30">

b0b3rt commented on packages/lesswrong/components/search/SearchPageTabbed.tsx:
```diff
@@ -469,6 +367,27 @@ const SearchPageTabbed = ({classes}:{
+          onClose={() => setModalOpen(false)}
+          aria-labelledby="search-filters-modal"
+          aria-describedby="search-filters-modal"
+          style={{display: modalOpen ? 'flex' : 'none'}}
```
Rather than apply inline styles, we use `classNames` to conditionally apply styles via className.  Search the codebase for `classNames(` for examples of how this works - it'd be something like `classNames(classes.filtersModal, { [classes.modalOpen]: modalOpen, [classes.modalClosed]: !modalOpen })` (and you'd need to define `modalOpen` and `modalClosed` in the `styles` for this component with `display: 'flex'` and `display: 'none'` respectively).

b0b3rt commented:
Appreciate the fix, and I'm fine with the icon - the MUI icon is indeed not amazing.  Looks good to me, pending addressing comments.

darkruby501 commented:
I tested it quickly and seems to work, regarding icons, yes, LW would use the following or similar:

<img width="440" alt="Material Icons - Material UI 2023-10-31 12-25-01" src="https://github.com/ForumMagnum/ForumMagnum/assets/7250541/aeb45bd7-b9d3-4ef9-9ecd-ab82ad9a1d88">

leoshatrushin commented:
Fixed the comments about using the palette and inline styles.

I thought I looked for the heroicon and it wasn't there, but I've now removed the custom svg and changed to use the existing icon for the EA forum.

The mui icon however, indeed doesn't seem to be there, so I used custom svg for that. @darkruby501 this is what it looks like
<img width="620" alt="Screenshot 2023-11-02 at 5 32 35 pm" src="https://github.com/ForumMagnum/ForumMagnum/assets/116286361/93bbd990-2780-4cb2-9338-224a0d5bff58">

Does the alignment forum typically follow LW choices?

oetherington commented:
Looks good to me - thanks @leoshatrushin!

Unfortunately, we're stuck on an ancient version of MUI which is why you couldn't find the icon there.

I'll approve, but wait for LW to confirm they're happy with the icon before merging.

jpaddison3 commented:
(For avoidance of doubt: FYI Leo cannot merge this as they don't have write access to this repo, so one of us (either @oetherington or the approving LW dev) will need to click merge.

b0b3rt commented:
Ruby's current offline - custom SVG looks good to me.  Double-checking that it looks correct in dark mode as well?  If so, all clear from the LW side.
